lib/core/macros.nim

```diff
@@ -1539,18 +1539,28 @@ proc customPragmaNode(n: NimNode): NimNode =
         for i in 0..<identDefsStack.len: identDefsStack[i] = obj[2][i]
         while identDefsStack.len > 0:
           var identDefs = identDefsStack.pop()
-          if identDefs.kind == nnkRecCase:
+
+          case identDefs.kind
+          of nnkRecList:
+            for child in identDefs.children:
+              identDefsStack.add(child)
+          of nnkRecCase:
+            # Add condition definition
             identDefsStack.add(identDefs[0])
-            for i in 1..<identDefs.len:
+            # Add branches
+            for i in 1 ..< identDefs.len:
               let varNode = identDefs[i]
-              # if it is and empty branch, skip
-              if varNode[0].kind == nnkNilLit: continue
-              if varNode[1].kind == nnkIdentDefs:
-                identDefsStack.add(varNode[1])
-              else: # nnkRecList
-                for j in 0 ..< varNode[1].len:
-                  identDefsStack.add(varNode[1][j])
-
+              case varNode.kind
+              of nnkOfBranch:
+                # We are only interested in variable definitions
+                if varNode[1].kind in {nnkIdentDefs, nnkRecList}:
```

Comment:
You need to use the last son of `varNode`, not `[1]` here.

Reply:
Then the dispatch can be removed and replaced with `varNode.last`

Reply:
I've replaced the dispatch code, like @alaviss mentioned. The last node is always of kind nnkRecList so it's processed the next iteration.